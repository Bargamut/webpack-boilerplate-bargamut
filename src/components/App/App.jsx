import * as React from 'react';

function App({ text = `Default title` }) {
  return (
    <h1>{text}</h1>
  );
}

export default App;
