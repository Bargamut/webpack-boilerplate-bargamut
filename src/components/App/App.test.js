'use strict';

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import App from './App';

const mock = {
  text: `Test data`,
};

it('should render App correctly without data', () => {
  const app = renderer
    .create(<App />)
    .toJSON();

  expect(app).toMatchSnapshot();
});

it('should render App correctly with mock title', () => {
  const { text: title } = mock;

  const app = renderer
    .create(<App text={title} />)
    .toJSON();

  expect(app).toMatchSnapshot();
});
