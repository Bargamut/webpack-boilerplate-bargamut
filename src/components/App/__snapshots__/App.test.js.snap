// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly with mock title 1`] = `
<h1>
  Test data
</h1>
`;

exports[`should render App correctly without data 1`] = `
<h1>
  Default title
</h1>
`;
